sw/host/opentitanlib/src/transport/proxy/spi.rs

```diff
@@ -143,4 +145,23 @@ impl Target for ProxySpi {
             _ => bail!(ProxyError::UnexpectedReply()),
         }
     }
+
+    fn assert_cs<'a>(self: Rc<Self>) -> Result<AssertChipSelect> {
+        match self.execute_command(SpiRequest::AssertChipSelect)? {
+            SpiResponse::AssertChipSelect => Ok(AssertChipSelect::create(self)),
+            _ => bail!(ProxyError::UnexpectedReply()),
+        }
+    }
+}
+
+impl TargetChipDeassert for ProxySpi {
+    fn deassert_cs(&self) {
+        match self
+            .execute_command(SpiRequest::DeassertChipSelect)
+            .expect("Error deactivating chip select")
+        {
+            SpiResponse::DeassertChipSelect => (),
+            _ => panic!("Error deactivating chip select"),
```

Comment:
It is correct that this is client code.  The problem is that this method has no means of returning an `Err()` as it is called from within a `drop()` method.
As such, the only option is to `panic!`, or to log something and continue.  However, if there is a communication problem with the server, as we are trying to release chip select, I believe that logging and continuing could lead to unexpected behavior, as further execution could go on, thinking that it repeatedly activates and deactivates chip select, when in reality, it is continuously asserted, due to the one request to release not being received by the session process, meaning that the count never again reaches zero.  This could potentially lead to the SPI target device mis-interpreting the data bytes sent to it, and performing operations that was not intended.  (E.g. if a write command first takes the address, and then any number of data bytes to be written on consecutive addresses, then missing a chip select deassertion could lead to the next commands being treated as further data bytes, overwriting areas that were not intended to be written to.)